module/zfs/vdev_raidz.c

```diff
@@ -1525,41 +1538,77 @@ vdev_raidz_io_start_write(zio_t *zio, raidz_row_t *rr, uint64_t ashift)
 {
 	vdev_t *vd = zio->io_vd;
 	raidz_map_t *rm = zio->io_vsd;
-	int c, i;
+	int c, i = 0, skip_first_cols = -1;
 
 	vdev_raidz_generate_parity_row(rm, rr);
 
-	for (int c = 0; c < rr->rr_cols; c++) {
+	IMPLY(rm->rm_nskip > 0, rm->rm_skipstart < rr->rr_scols);
+
+	if (rr->rr_scols < (rm->rm_skipstart + rm->rm_nskip)) {
+		skip_first_cols =
+		    (rm->rm_skipstart + rm->rm_nskip) % rr->rr_scols;
+	}
+
+	ASSERT3S(skip_first_cols, <, rr->rr_scols);
+	IMPLY(rm->rm_nskip == 0, skip_first_cols = -1);
```

Comment:
If above we agreed that rr->rr_scols < rm->rm_skipstart, then if rm->rm_nskip == 0, then rr->rr_scols < (rm->rm_skipstart + rm->rm_nskip) and skip_first_colsis never assigned, so == -1.  So this check just not needed.  Which is good, because you've used "=" instead of "==".

Reply:
Agreed, I had a lot of this in there just for sanity checks while developing to make sure I did not make a mistake, but it can be removed. Ughhhh... Yeah, doing `=` instead of `==` was not good... I removed it.